src/pages/postgraphile/usage-schema.md

```diff
@@ -238,3 +238,288 @@ look at the `postgraphile-core` and `graphile-build-pg` modules.
+`./src/graphql/getMovies.graphql`.
+
+```graphql
+query GetMovies($top: Int!) {
+  movies(first: $top) {
+    nodes {
+      id
+      title
+    }
+  }
+}
+```
+
+Save the file and run the code generation task.
+
+Alternatively we can create some inline query directly in code like this:
+
+```js
+import { makeExtendSchemaPlugin, gql } from "graphile-utils";
+
+// inside some function:
+
+const GetMoviesDocument = gql`
+  query Query {
+    __typename
+    movies(first: 3) {
+      nodes {
+        id
+        title
+      }
+    }
+  }
+`;
+```
+
+The query can then be used in your code via the generated type or inline:
+
+```js
+
+  // This would be generated from the above .graphql file by the code generator
+  // alternatively define it inline via gql` like shown above
+  const doc = GetMoviesDocument;
+  const variables: GetMoviesQueryVariables = {
+    top: 3,
+  };
+
+  // Execute the query from your backend code
+  // The "Query" type is your the root Query type. It contains all the query
+  // endpoints from the GraphQL API with type support
+  const queryResult = await graphql.execute<Query>(
+    resolveInfo.schema,
+    doc,
+    undefined,
+    context,
+    variables,
+  );
+
+  if (queryResult.errors) {
+    // do something in error case
+  }
+
+  // the result can then be used to get the returned data
+  const allTitles = queryResult.data?.movies?.nodes.map(
+    (movie: any) => movie.title,
+  );
+```
+
+### Usage in different resolvers
+
+To use the GraphQL inside of our backend we need to get access to the "graphql"
+variable. This variable is available for example in the following places:
+
+```js
+export const BusinessLogicPlugin = makeExtendSchemaPlugin((build: Build) => {
+  const { graphql } = build;
+```
+
+```js
+export const WrapResolverPlugin = makeWrapResolversPlugin({
+  Mutation: {
+    async updateMovieById(
+      resolve: any,
+      source: any,
+      args: { [argName: string]: any },
+      context: any,
+      resolveInfo: GraphQLResolveInfo,
+    ) {
+    const query = context.pgClient.query;
+```
+
+### Build-time validation of inline documents
+
+If you want to write inline
+gql`queries it is possible to validate them during build-time. Adjust the created`codegen.yml`
+file with the following section at the same level of the existing
+"src/generated/types.ts:" part.
+
+This will create types from any inline defined document. As we do not actually
+need the files but only use them for build-time validation of our inline queries
+we can have the types be put to some "trash" location.
+
+```yaml
+# Creates types for inline gql`...` defined queries.
+# This is only used for build-time validation of those queries
+trash/inlineGqlValidation.ts:
+  documents: "src/**/*.ts"
+  plugins:
+    - typescript
+    - typescript-operations
+    - typescript-urql
+  config:
+    withHOC: false
+    withComponent: false
+    withMutationFn: false
+```
```

Comment:
I think we should delete this section. It's better to use something like `eslint-plugin-graphql` to do validation if you don't intend to use the types - it has better integration with editors/CI/etc.

```suggestion
```

Reply:
Thanks for the tip - will look into `eslint-plugin-graphql`.